### DISCUSSION TITLE
Monitoring

### DISCUSSION THREAD

**Discussion by jenslink** [becarpenter/book6#68] [General]
There should be something on security and availability monitoring. Maybe even a whole chapter?

**Comment by becarpenter** [comment]
There's already an empty section at 6. Management and Operations/Security operation.md . I'm not sure what we should put there.
And yes, a section on availability monitoring is needed in that chapter too. If a topic gets too big for a section, we can always convert it into a chapter later.

Do you think there is much that is specific to IPv6 here? Obviously the fine details are different, but are there any high-level differences?

**Reply by jenslink** [reply]
From what I see is that IPv6 monitoring is often forgotten, ignored or done from the wrong vantage point. 

Some examples: 
- Large corporate network with no IPv6 monitoring because the part of the network where the monitoring system was located had no IPv6. 
- Web services with AAAA records and proper configuration, monitoring said that everything okay, but you couldn't access the web services via IPv6 from the internet. Someone forgot a firewall rule. And the monitoring system was on the inside of the network. 
- Mail (SMTP) server with AAAA records, but IPv6 was disabled (or blocked by a firewall?) for whatever reason on the server, but nobody removed the AAAA records. Wasn't noticed internally, my guess: They did not monitor via IPv6.  

I don't think there is no big difference between monitoring services for IPv4 or IPv6 it just has to be done for all services and, if they are dual-stacked, for both protocols. 

In case of the mail sever example there were probably three different teams involved and they either didn't talk to each other or that some bad process implemented and no automation. 

I like the idea that implementing IPv6 also gives you the chance to clean up your  documentation, your infrastructure and your processes. And may also be a way to implement automation.

**Comment by becarpenter** [comment]
All sounds true. If you can turn that into a book section that would be great.

**Comment by becarpenter** [comment]
There's now a [monitoring section](https://github.com/becarpenter/book6/blob/main/6.%20Management%20and%20Operations/Benchmarking%20and%20monitoring.md).
I copy-edited Jens's text as a start, but PRs are welcome!
